e2e/e2e-suite_test.go

```diff
@@ -47,3 +51,21 @@ var _ = BeforeSuite(func() {
 	// parse test suite arguments
 	flag.Parse()
 })
+
+var _ = JustAfterEach(func() {
+	if CurrentSpecReport().Failed() {
+		dump(os.Getenv("KUBECONFIG"), "")
+	}
+})
+
+func dump(kubeconfig, artifactsSuffix string) {
+	cmd := exec.Command("hack/dump.sh", "--kubeconfig", kubeconfig)
```

Comment:
i'd rather that the e2e tests take an argument that points to the path of the dump binary up front as a precondition to running the test suite.

So, If the dump binary path is provided, then execute it here, otherwise skip.